README.md

```diff
@@ -39,38 +39,72 @@ I've created [an example project](https://github.com/evollu/react-native-fcm/tre
 ### FCM config file
 
 In [firebase console](https://console.firebase.google.com/), you can:
-- for **Android**: download `google-services.json` file and place it in `android/app` directory
+- for **Android**: download `google-services.json` file and place it in `android/app` directory.
 - for **iOS**: download `GoogleService-Info.plist` file and place it in `/ios/your-project-name` directory (next to your `Info.plist`)
 
 Make sure you have certificates setup by following
 https://firebase.google.com/docs/cloud-messaging/ios/certs
 
+---
+
 ## Android Configuration
 
 - As `react-native link` sometimes has glitches, make sure you have this line added
 
-https://github.com/evollu/react-native-fcm/blob/master/Examples/simple-fcm-client/android/app/src/main/java/com/google/firebase/quickstart/fcm/MainApplication.java#L28
+### Edit `android/build.gradle`:
+
+**NOTE:** The followed line may not be up-to-dated. Please refer to https://firebase.google.com/docs/android/setup
 
-- Edit `android/build.gradle`:
-```diff
-  dependencies {
-    classpath 'com.android.tools.build:gradle:2.0.0'
-+   classpath 'com.google.gms:google-services:3.0.0'
 ```
+buildscript {
+    repositories {
+        // ...
+        google() // Google's Maven repository
+    }
+    // ...
+    dependencies {
+        // ...
+        classpath 'com.google.gms:google-services:4.0.2' // google-services plugin
+    }
+}
 
-- Edit `android/app/build.gradle`. Add at the bottom of the file:
-```diff
-  apply plugin: "com.android.application"
-  ...
-+ apply plugin: 'com.google.gms.google-services'
+allprojects {
+    // ...
+    repositories {
+        // ...
+        google() // Google's Maven repository
+    }
+}
 ```
 
-- Edit `android/app/src/main/AndroidManifest.xml`:
+### Edit `android/app/build.gradle`:
+
+**NOTE:** Please refer to https://firebase.google.com/docs/android/setup
+
+```
+dependencies {
+    // ...
+    compile project(':react-native-fcm')
+
+    // ...
+    implementation 'com.google.firebase:firebase-core:16.0.1'
```

Comment:
where does this come from? why does it have different version between core and messaging?

Reply:
we support version using variables. I think that is the way other packages doing it as well. Can we go with that approach in the readme?
```
def DEFAULT_FIREBASE_CORE_VERSION           = "12.+"
def DEFAULT_FIREBASE_MESSAGING_VERSION      = "12.+"
```

Reply:
Just reverify a bit these lines, realised that should modify the src for not using these DEFAULT version instead. The "12.+" limits the version to 12.x.x which is too old for now. I will commit a new one again